packages/base-controller/src/RestrictedControllerMessenger.ts

```diff
@@ -291,15 +299,59 @@ export class RestrictedControllerMessenger<
    * @param event - The event type. This is a unique identifier for this event.
    * @template EventType - A type union of Event type strings that are namespaced by Namespace.
    */
-  clearEventSubscriptions<EventType extends NamespacedName<Namespace>>(
-    event: EventType,
-  ) {
+  clearEventSubscriptions<
+    EventType extends Event['type'] & NamespacedName<Namespace>,
+  >(event: EventType) {
     /* istanbul ignore if */ // Branch unreachable with valid types
-    if (!event.startsWith(`${this.#controllerName}:`)) {
+    if (!this.#isInCurrentNamespace(event)) {
       throw new Error(
         `Only allowed clearing events prefixed by '${this.#controllerName}:'`,
       );
     }
     this.#controllerMessenger.clearEventSubscriptions(event);
   }
+
+  /**
+   * Determine whether the given event type is allowed. Event types are
+   * allowed if they are in the current namespace or on the list of
+   * allowed events.
+   *
+   * @param eventType - The event type to check.
+   * @returns Whether the event type is allowed.
+   */
+  #isAllowedEvent(eventType: Event['type']): eventType is AllowedEvent {
+    // Safely upcast to allow runtime check
```

Comment:
Oh, good trick.

Reply:
Yeah I thought so! I was considering suggesting this strategy in our docs.

That is, if you want to upcast something (which is always safe to do), assign it to a new variable. This lets TypeScript validate that it's a safe upcast.

Whereas the `as` operator also allows downcasting, so it's not as safe to use.